### ISSUE TITLE
Everburn Urn - ticking block entity

### ISSUE THREAD

**Issue by sephlington** [nekosune/Thaumic_Exploration#40]
Everburn Urn crashing the game with a stack overflow. It's being fed ignis vis through Thaumic Energistic's Vis Relay Interface. Almost straight after the vis relay was placed and the vis started channelling into the Urn, the game crashed. Reopening crashed again, as seen in the crash log below. I had to temporarily uninstall the mod to get into my world again.

Crash log: http://pastebin.com/ziQvt8n4

Cross-linking from: https://github.com/Flaxbeard/Thaumic_Exploration/issues/58

**Comment by Technus** [comment]
Urn method uses wrong method from TC API

https://github.com/nekosune/Thaumic_Exploration/blob/master/src/main/java/flaxbeard/thaumicexploration/tile/TileEntityEverburnUrn.java#L146

From TC API:
     \* This (consumeVis) method should never be called directly. Use VisNetHandler.drainVis() instead
     \* @param aspect what aspect to drain
     \* @param vis how much to drain
     \* @return how much was actually drained

**Comment by Technus** [comment]
The code should be more like:

```
...
if(this.ticks==10) {
        if(this.ignisVis<16){
                ignisVis += VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.FIRE, 1);
                worldObj.markBlockForUpdate(xCoord,yCoord,zCoord);
        }    
        ticks=0;
}
...
```

**Comment by Walkman100** [comment]
I'm getting this issue on MC 1.7.10 with Thaumcraft 4.2.3.5, Thaumic Exploration 1.1.53, Thaumic Energistics 1.0.0.5 and a bunch of other mods I don't think are related. ([OpenEye crash](http://openeye.openmods.info/crashes/c5569669872fd54def9e79e2a9432fc6)) 

Does anyone know if there's anyway to fix this other than removing the urn?

**Comment by Technus** [comment]
@Walkman100 version used in GTNH pack should work.

**Comment by Walkman100** [comment]
Yeah I'm using v1.1.57 from [here](https://github.com/GTNewHorizons/Thaumic_Exploration/releases/tag/1.1.57) now, one thing I noticed: it still acts as a vis relay, but it pulls ignis like any other thaum machine (Infernal Furnace, Focal Manipulator e.t.c.), which I see is as per this comment: https://github.com/GTNewHorizons/NewHorizons/issues/681#issuecomment-242234283. I made a comment there too, but thanks very much for the fix 👍
